### PR TITLE
Add bitmask to indicate EKF logic state

### PR DESCRIPTION
**Pull request by jiejiezhang**
add bitmask to indicate EKF logic state

### CODE REVIEW

**Comment by jiejiezhang** [comment]
@julianoes  I don't find the reason of CI error. Could you help to see?

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4189,6 +4244,7 @@
...
       <field type="uint64_t" name="time_usec" units="us">Timestamp (micros since boot or Unix epoch)</field>
       <field type="uint16_t" name="flags" enum="ESTIMATOR_STATUS_FLAGS" display="bitmask">Integer bitmask indicating which EKF outputs are valid. See definition for ESTIMATOR_STATUS_FLAGS.</field>
+      <field type="uint16_t" name="mode_flags" enum="ESTIMATOR_MODE_FLAGS" display="bitmask">Integer bitmask indicating which valid data of ekf is fused. See definition for ESTIMATOR_MODE_FLAGS.</field>
```
Comment (julianoes) [inline message_definitions/v1.0/common.xml]: It is not possible to add a new field inside the message. However, you can add this field at the end of the message after a tag `<extensions/>`.

Comment (julianoes) [reply message_definitions/v1.0/common.xml]: @priseborough: Do you think 16 bits is enough for the future?

**Comment by julianoes** [comment]
@jiejiezhang CI fails, can you check?

**Comment by jiejiezhang** [comment]
Sorry. Now i'm still not find the ci error reason.

**Comment by julianoes** [comment]
@jiejiezhang this took me a while to figure out. The last enum field is a duplicate, that's why it fails.

**Comment by jiejiezhang** [comment]
> this took me a while to figure out. The last enum field is a duplicate, that's why it fails.

Sorry for that my mistake. Thanks!!!

**Comment by amilcarlucas** [comment]
if nobody objects, I will merge this within one week

**Comment by TSC21** [comment]
@amilcarlucas I actually think this should go through [RFCs](https://github.com/mavlink/rfcs) first. This PR was launched much later than https://github.com/mavlink/mavlink/pull/667, and turns out to be concurrent with it. https://github.com/mavlink/mavlink/pull/667 was replaced by https://github.com/mavlink/mavlink/pull/880 in the mean time. https://github.com/mavlink/mavlink/pull/880 is a more general fashion way of handling the estimator status, while this PR seems to be too much directed to the EKF estimators onboard the flight controllers.

**Comment by hamishwillee** [comment]
Closing. No movement for 4 years means to me that either there is no compelling need or it has been superseded. If someone wants to champion the underlying functionality happy to take a clean PR that links back to this for discussion.

**Pull request closed without merging**